calculator.c

```diff
@@ -0,0 +1,141 @@
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+int numberStack[50];
+int nSTop = -1;
+int operatorStack[50];
+int opSTop = -1;
+
+int evaluateExp(char exp[]);
+void calculate();
+int operatorPrecedence(char operator);
+int custom_isdigit(char ch);
+int custom_isspace(char ch);
+
+int main()
+{
+    char exp[80];
+    int result = 0;
+    fgets(exp, sizeof(exp), stdin);
+    exp[strcspn(exp, "\n")] = '\0';
+    if (strlen(exp) == 0)
+    {
+        printf("Error: Invalid expression.\n");
+        return 0;
+    }
+    result = evaluateExp(exp);
+    printf("%d\n", result);
+    return 0;
+}
+
+int evaluateExp(char exp[])
+{
+    for (int i = 0; i < strlen(exp); i++)
+    {
+        char ch = exp[i];
+        if (custom_isdigit(ch))
+        {
+            int num = 0;
+            while (custom_isdigit(ch))
+            {
+                num = num * 10 + (ch - '0');
+                i++;
+                if (i < strlen(exp))
+                {
+                    ch = exp[i];
+                }
+                else
+                {
+                    break;
+                }
+            }
+            i--;
+            if (nSTop >= 49)
+            {
+                printf("Error: Invalid expression.\n");
+                exit(0);
+            }
+            numberStack[++nSTop] = num;
+        }
+        else if (ch == '+' || ch == '-' || ch == '/' || ch == '*')
+        {
+            while (opSTop != -1 && operatorPrecedence(ch) <= operatorPrecedence(operatorStack[opSTop]))
+            {
+                calculate();
+            }
+            if (opSTop >= 49)
+            {
```

Comment:
you are rewriting printf("Error: Invalid expression.\n");exit(0); multiple times in the code, cannot we create function for this? What if I update the error message, will we go and change at all the places?

Reply:
I create a single errorHandler function to handle all error scenarios.